HomeWork1/Novikov_Homework_2.cpp

```diff
@@ -0,0 +1,283 @@
+#include <cmath>
+#include <iostream>
+
+void Task_1_average_numbers();
+void Task_2_lucky_ticket();
+void Task_3_number_reverce();
+void Task_4_odd_numbers();
+void Task_5_best_divider();
+void Task_6_christmas_tree();
+void Task_7_bit_count();
+void Task_8_bit_set();
+void Task_9_number_builder();
+
+int main() {
+
+  std::cout << "Homework task one average of digits\n";
+  Task_1_average_numbers();
+
+  std::cout << "Homework task two lucky ticket\n";
+  Task_2_lucky_ticket();
+
+  std::cout << "Homework task three reverse number\n";
+  Task_3_number_reverce();
+
+  std::cout << "Homework task four sum of odd elements\n";
+  Task_4_odd_numbers();
+
+  std::cout << "Homework task five the best divider\n";
+  Task_5_best_divider();
+
+  std::cout << "Homework task six christmas tree\n";
+  Task_6_christmas_tree();
+
+  std::cout << "Homework task seven number of set bits\n";
+  Task_7_bit_count();
+
+  std::cout << "Homework task eight check state of bit number\n";
+  Task_8_bit_set();
+
+  std::cout << "Homework task nine number building\n";
+  Task_9_number_builder();
+}
+
+int sum_counter(int number) {
+  int result{0};
+  while (number) {
+    result += (number % 10);
+    number /= 10;
+  }
+  return result;
+}
+
+void Task_1_average_numbers() {
+  int number{0};
+  std::cout << "Enter your number: ";
+  std::cin >> number;
+
+  if (number == 0) {
+    std::cout << "Number is zero";
+    return;
+  }
+
+  const int numberOfDigits = static_cast<int>(log10(number)) + 1; // math magic
+
+  if (numberOfDigits == 1) {
+    std::cout << "Average is " << number << std::endl;
+    return;
+  }
+
+  float average{0.0};
+  while (number) {
+    average += (number % 10) / static_cast<float>(numberOfDigits);
+    number /= 10;
+  }
+
+  std::cout << "Average is " << average << std::endl;
+}
+
+void Task_2_lucky_ticket() {
+  bool check{true};
+  int number;
+  int digitsNumber{0};
+  while (check) {
+    std::cout << "Enter 6-digit number" << std::endl;
+    std::cin >> number;
+    if (number == 0) {
+      std::cout << "Number is zero\n";
+      continue;
+    }
+    digitsNumber = static_cast<int>(log10(number)) + 1;
+
+    if (digitsNumber != 6) {
+      std::cout << "Number is not 6-digit! Please try again!\n";
+      continue;
+    } else
+      check = false;
+  }
```

Comment:
```suggestion
    if (digitsNumber == 6) {
      break;
    }
      std::cout << "Number is not 6-digit! Please try again!\n";
```
я думаю что булевая переменная там лишняя, но покраней мере может без нее обойтись